lib/ansible/galaxy/role.py

```diff
@@ -303,54 +362,16 @@ def install(self):
 
-                # we strip off any higher-level directories for all of the files contained within
-                # the tar file here. The default is 'github_repo-target'. Gerrit instances, on the other
-                # hand, does not have a parent directory at all.
-                installed = False
-                while not installed:
-                    display.display("- extracting %s to %s" % (self.name, self.path))
+                for path in self.paths:
+                    self.path = path
                     try:
-                        if os.path.exists(self.path):
-                            if not os.path.isdir(self.path):
-                                raise AnsibleError("the specified roles path exists and is not a directory.")
-                            elif not context.CLIARGS.get("force", False):
-                                raise AnsibleError("the specified role %s appears to already exist. Use --force to replace it." % self.name)
-                            else:
-                                # using --force, remove the old path
-                                if not self.remove():
-                                    raise AnsibleError("%s doesn't appear to contain a role.\n  please remove this directory manually if you really "
-                                                       "want to put the role here." % self.path)
-                        else:
-                            os.makedirs(self.path)
-
-                        # now we do the actual extraction to the path
-                        for member in members:
-                            # we only extract files, and remove any relative path
-                            # bits that might be in the file for security purposes
-                            # and drop any containing directory, as mentioned above
-                            if member.isreg() or member.issym():
-                                n_member_name = to_native(member.name)
-                                n_archive_parent_dir = to_native(archive_parent_dir)
-                                n_parts = n_member_name.replace(n_archive_parent_dir, "", 1).split(os.sep)
-                                n_final_parts = []
-                                for n_part in n_parts:
-                                    if n_part != '..' and '~' not in n_part and '$' not in n_part:
-                                        n_final_parts.append(n_part)
-                                member.name = os.path.join(*n_final_parts)
-                                role_tar_file.extract(member, to_native(self.path))
-
-                        # write out the install info file for later use
-                        self._write_galaxy_install_info()
-                        installed = True
+                        self._extract(role_tar_file, archive_parent_dir)
+                        break
```

Comment:
Is this an equivalent of the installed check from the original while-loop? This may deserve a code comment explaining the motivation. Also, I'd move it to the `else` block (right after `except`) so it'd be easier to read as "if try block succeeds, proceed and interrupt that loop".